frontend/src/metabase/visualizations/visualizations/Sankey.jsx

```diff
@@ -0,0 +1,212 @@
+/* eslint-disable react/prop-types */
+import React, { Component } from "react";
+import ReactDOM from "react-dom";
+import { t } from "ttag";
+import d3 from "d3";
+import cx from "classnames";
+
+import _ from "underscore";
+
+import { isNumeric } from "metabase/lib/schema_metadata";
+
+import { sankey, sankeyLinkHorizontal } from "d3-sankey";
+import chroma from "chroma-js";
+
+import {
+  SANKEY_DATA_SETTINGS,
+} from "../lib/settings/graph";
+
+const MAX_WIDTH = 500;
+const PADDING_BOTTOM = 10;
+
+export default class Sankey extends Component {
+  static uiName = t`Sankey`;
+  static identifier = "sankey";
+  static iconName = "sankey";
+
+  static minSize = { width: 4, height: 4 };
+
+  static isSensible({ cols, rows }) {
+    return rows.length === 1 && cols.length === 1;
+  }
+
+  static checkRenderable([
+    {
+      data: { cols, rows },
+    },
+  ]) {
+    if (!isNumeric(cols[0])) {
+      throw new Error(t`Sankey visualization requires a number.`);
+    }
+  }
+
+  state = {
+    mounted: false,
+  };
+
+  _label;
+
+  static settings = {
+    ...SANKEY_DATA_SETTINGS,
+  };
+
+  componentDidMount() {
+    this.setState({ mounted: true });
+  }
+  componentDidUpdate() {
+  }
+  
+  getFilteredData(cols, rows, dimensions, metrics){
+    const valueIndex = _.findIndex(
+      cols,
+      col => col.name === metrics[0],
+    );
+    var dims = []
+    for(let i=0; i<dimensions.length; i++){
+      dims.push(
+        _.findIndex(
+          cols,
+          col => col.name === dimensions[i],
+        )
+      )
+    }
+
+    let nodes = []
+    rows.map((row) => {
+      for(let i=0; i<dimensions.length; i++){
+        if(nodes.findIndex(x => x.name==row[dims[i]]) === -1){
+          nodes.push({"name": row[dims[i]]})
+        }
+      }
+    });
+    
+    let links = []
+    for(let i=0; i<dimensions.length-1; i++){
+      rows.map((x, index) => {
+        links.push({
+          "target": _.findIndex(
+            nodes,
+            n => n.name === x[dims[i]],
+          ), 
+          "source": _.findIndex(
+            nodes,
+            n => n.name === x[dims[i+1]],
+          ),
+          "value": x[valueIndex]
```

Comment:
For consistency with other charts, this could use the help of `formatValue` (from `metabase/lib/formatting`).

Reply:
@morgan12 did you try using `formatValue` here?